### DISCUSSION TITLE
DbSnapshot identifier property returns full ARN

### DISCUSSION THREAD

**Discussion by celaus** [awslabs/aws-sdk-rust#805] [General]
Hi! It looks like the `db_snapshot_identifier` of the `DbSnapshot` property returns the ARN of the snapshot instead of the identifier. This makes it awkward to pass into another API call, resulting in the following error. Obviously I can get the snapshot identifier from the ARN but since there is a property `db_snapshot_arn`, I thought this may be a bug? 

```2023-05-11T14:15:50.407827Z DEBUG ThreadId(10) describe_db_snapshots_attrs{snapshot_id="arn:aws:rds:us-east-1:123123456456:snapshot:my-bucket-name"}:send_operation{operation="DescribeDBSnapshotAttributes" service="rds" status="service_error" service error: unhandled error: unhandled error: Error { code: "InvalidParameterValue", message: "The parameter DBSnapshotIdentifier is not a valid identifier. Identifiers must begin with a letter; must contain only ASCII letters, digits, and hyphens; and must not end with a hyphen or contain two consecutive hyphens.", ... ``` 

What say you? 

SDK version `0.27.0`

**Comment by Velfi** [comment]
Hey @celaus, thanks for submitting this issue.

@jmklix Can you move this into the cross-SDK issue tracker?

**Comment by Velfi** [comment]
@celaus There's likely nothing our team can do about this other than to submit a ticket upstream. We'll do that, but things like this often take a long time to change, if they ever do.
